include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
Exceptions are bubbling up the call stack until the first `catch` that can handle them. This means the `try { ... } catch () { ... }` block can be in `index.php` or wherever the logging system has already been initialized. This ensures that all Exceptions will be caught at least by the top-level `catch` block. Specific `try {} catch () {}` blocks can be added if there's a specific behavior to enable in case of extraordinary failure.

What I mean is that Exceptions should only be raised when the function cannot return the expected data. In this case, the expected return data is the sanitized gcontact array. However, if there is an empty URL, then the function cannot return the expected data, and should throw an Exception instead of simply returning `false`.

All of this allows you to be almost certain that every function returns the expected data type instead of having to constantly type-check every single return value. I say almost because you can still hit a snag in the function you didn't cover with Exception, but ideally every function should also check the input parameters type and throw an according Exception.